### PR TITLE
refactor(image): convert to typescript

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour

All component files should be refactored and converted to TypeScript.

### Current behaviour

Component files are currently javascript.

### Checklist

- [x] Commits follow our style guide
~~- [ ] Related issues linked in commit messages if required~~
~~- [ ] Screenshots are included in the PR if useful~~
~~- [ ] All themes are supported if required~~
- [x] Unit tests added or updated if required
~~- [ ] Cypress automation tests added or updated if required~~
~~- [ ] Storybook added or updated if required~~
~~- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required~~
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/image/image.component.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import invariant from "invariant";
+import { StyledImage, StyledImageProps } from "./image.style";
+
+export const Image = ({ alt, src, children, ...rest }: StyledImageProps) => {
+  invariant(
+    !src || (src && !children),
```
Comment (robinzigmond) [inline src/components/image/image.component.tsx:7]:
```suggestion
    !src || !children,
```

Comes to the same thing but imo it's easier to understand this way.

File: src/components/image/image.component.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "The 'Image' component renders as an 'img' element when the 'src' prop is used and therefore does not accept children."
+  );
+
+  invariant(
+    !src || (src && alt && typeof alt === "string"),
```
Comment (robinzigmond) [inline src/components/image/image.component.tsx:12]:
```suggestion
    !src || (alt && typeof alt === "string"),
```

same as above

File: src/components/image/image.style.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+`;
+
+StyledImage.defaultProps = {
+  backgroundPosition: "center",
+  backgroundRepeat: "no-repeat",
+  backgroundSize: "cover",
+  theme: baseTheme,
+};
+
+// eslint-disable-next-line import/prefer-default-export
```
Comment (robinzigmond) [inline src/components/image/image.style.ts:46]: I don't think we need to disable eslint here as we're also exporting the `StyledImageProps` (I'm not seeing any errors/warnings when I remove the comment).

File: src/components/image/image.spec.tsx [hunk 4]
```diff
@@ -29,36 +29,33 @@ describe("Image", () => {
...
+        );
+      }).toThrow(errorMessage);
     });
 
     it("throws an error when `src` is passed and no `alt` provided", () => {
-      mount(<Image src="foo.jpg" />);
-      expect(console.error).toHaveBeenCalled();
-    });
-
-    it("does not throw an error when `src` is passed and `alt` provided", () => {
```
Comment (robinzigmond) [inline src/components/image/image.spec.tsx:54]: is there a good reason for removing this test? Why not something like `expect(() => { mount(<Image src="foo.jpg" alt="foo"/>); }).not.toThrow();`?

**Review by edleeks87 — APPROVED** [review]

File: src/components/image/image.component.tsx (hunk 1 above)
Comment (grabkowski) [reply src/components/image/image.component.tsx:7]: right, I got carried away when reversing that conditional for invariant

File: src/components/image/image.spec.tsx (hunk 4 above)
Comment (robinzigmond) [reply src/components/image/image.spec.tsx:54]: Oh yes, of course. I withdraw the comment then :)

**Review by robinzigmond — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**